spec/controllers/redirections_controller_spec.rb

```diff
@@ -10,6 +10,20 @@
 
       expect(response).to redirect_to(next_redirection.url)
     end
+
+    it "creates a Redirection if it doesn't exist" do
+      new_slug = "new"
+      url = "http://example.com"
+      old_next_url = Redirection.first.next_url
+
+      request.env["HTTP_REFERER"] = url
+      get :next, slug: new_slug
+
+      expect(Redirection.first.next.slug).to eq new_slug
```

Comment:
Maybe put some of these `Redirection`s into variables? Some of the method chaining is hard to read.